Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Pointers/SpherePointer.cs

```diff
@@ -257,14 +268,81 @@ public bool TryUpdateQueryBufferForLayerMask(LayerMask layerMask, Vector3 pointe
 
                 for (int i = 0; i < numColliders; i++)
                 {
-                    grabbable = queryBuffer[i].GetComponent<NearInteractionGrabbable>();
+                    Collider collider = queryBuffer[i];
+                    grabbable = collider.GetComponent<NearInteractionGrabbable>();
+                    // Additional check: is grabbable in the camera frustrum
                     if (grabbable != null)
+                    {
+                        if (!isInFOV(collider))
+                        {
+                            grabbable = null;
+                        }
+                    }
+                    if (grabbable != null)
+                    {
+                        return true;
+                    }
+                }
+                return false;
+            }
+
+
+            /// <summary>
+            /// Returns true if a collider's bounds is within the camera FOV
+            /// </summary>
+            /// <param name="myCollider">The collider to test</param>
+            /// <returns></returns>
+            private bool isInFOV(Collider myCollider)
+            {
+                Bounds b = myCollider.bounds;
+                var corners = new List<Vector3>();
+                BoundsExtensions.GetColliderBoundsPoints(myCollider, corners, 0);
+                float xMin = float.MaxValue, yMin = float.MaxValue, zMin = float.MaxValue;
+                float xMax = float.MinValue, yMax = float.MinValue, zMax = float.MinValue;
+                foreach (var pt in corners)
+                {
+                    if (isPointInFrustrum(pt))
                     {
                         return true;
                     }
+
+                    xMin = Mathf.Min(xMin, pt.x);
+                    yMin = Mathf.Min(yMin, pt.y);
+                    zMin = Mathf.Min(zMin, pt.z);
+                    xMax = Mathf.Max(xMax, pt.x);
+                    yMax = Mathf.Max(yMax, pt.y);
+                    zMax = Mathf.Max(zMax, pt.z);
+                }
+
+                // edge case: check if camera is inside the entire bounds of the collider;
```

Comment:
Looking at the test case - is there one also for this edge case?

i.e. I see a comment around this one:

// Move forward so that cube is no longer visible

But it looks like it's actually a 'false' case.

It's possible I'm not reading deeply into this enough.

Reply:
at the cube no longer visible test we're moving past the object, so it's not in FOV anymore. The edge case is if you have an object that is so big that it spans the entire FOV, so a cube in front of you that takes up the entire screenspace. We could add a test for that by just spawning a huge cube and checking if the pointer is active?

Reply:
sry in front of you is wrong - it has to surround you as well.

though now thinking about this - if we have that scenario, do we really want to leave that pointer enabled? like if there is a gigantic cube surrounding us like a room and we're close to the back wall of that room and there's objects to manipulate inside of that cube that might be able to trigger far interaction? 

On the other hand , not sure if you would want to define a thing like that as a grabbable then..... probably not

Reply:
>though now thinking about this - if we have that scenario, do we really want to leave that pointer enabled? like if there is a gigantic cube surrounding us like a room and we're close to the back wall of that room and there's objects to manipulate inside of that cube that might be able to trigger far interaction?

Actually line 56 tests exactly this case -- you are inside of a giant grabbable. I considered not supporting this edge case but then remembered a customer that mentione he was pleasantly surprised that he could move objects when he was inside of them. So if he accidentally was inside a huge object he could still move it. So I decided to support that edge case explicitly.

Reply:
>Looking at the test case - is there one also for this edge case?

Yes, see SpherePointerTests line 56.

Reply:
@thalbern  could you let me know if my comments resolve your concerns here?

Reply:
yes, i didn't realize the cube was surrounding us in that case - all good :)